### PR TITLE
Add 'remove' button when wallet is locked #269

### PR DESCRIPTION
**Pull request by thatStrangeGuyThorchain**
- [x] added remove button

closes https://github.com/thorchain/asgardex-electron/issues/269

### CODE REVIEW

File: src/views/wallet/UnlockView.tsx [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+import { initialKeystoreState } from '../../services/wallet/service'
 
 const UnlockView: React.FC = (): JSX.Element => {
   const { keystoreService } = useWalletContext()
-  const keystore = useObservableState(keystoreService.keystore$, none)
+  const keystore = useObservableState(keystoreService.keystore$, initialKeystoreState())
+  const history = useHistory()
+
+  useEffect(() => {
+    const shouldPass = Boolean(toNullable(keystore))
```
Comment (veado) [inline src/views/wallet/UnlockView.tsx:18]: Just curious: Why is `Boolean` needed here?

File: src/services/wallet/types.ts [hunk 2]
```diff
@@ -7,12 +7,18 @@ export type Phrase = string
...
-export type KeystoreContent = { phrase: Phrase }
-export type KeystoreState = Option<Option<KeystoreContent>>
+export enum WalletStatus {
+  LOCKED = 'LOCKED',
+  UNLOCKED = 'UNLOCKED'
+}
+
+export type KeystoreContent = { status: WalletStatus.LOCKED } | { status: WalletStatus.UNLOCKED; phrase: Phrase }
+
+export type KeystoreState = Option<KeystoreContent>
```
Comment (veado) [inline src/services/wallet/types.ts:21]: As mentioned in Slack, I'm not happy about this change: More code, it includes redundant information. IMO we don't need this change.

File: src/i18n/de/common.ts [hunk 3]
```diff
@@ -13,7 +13,8 @@ const common: CommonMessages = {
   'common.coin': 'Coin - DE',
   'common.password': 'Passwort',
   'common.memo': 'Memo - DE',
-  'common.refresh': 'aktualisierung'
+  'common.refresh': 'aktualisierung',
+  'common.remove': 'verwijderen'
```
Comment (veado) [inline src/i18n/de/common.ts:17]: ^ Entfernen

File: src/i18n/de/common.ts [hunk 4]
```diff
@@ -13,7 +13,8 @@ const common: CommonMessages = {
   'common.coin': 'Coin - DE',
   'common.password': 'Passwort',
   'common.memo': 'Memo - DE',
-  'common.refresh': 'aktualisierung'
+  'common.refresh': 'aktualisierung',
```
Comment (veado) [inline src/i18n/de/common.ts:16]: ^ Aktualisierung

File: src/components/wallet/UnlockForm.tsx [hunk 5]
```diff
@@ -83,6 +88,15 @@ const UnlockForm: React.FC<Props> = (props: Props): JSX.Element => {
     form.resetFields()
   }, [form])
 
+  const onRemove = useCallback(() => {
+    const isConfirmed = window.confirm('remove wallet?')
+    if (isConfirmed) {
+      removeKeystore().then(() => {
```
Comment (veado) [inline src/components/wallet/UnlockForm.tsx:94]: Do we really need `.then` handler here ? `removeKeystore()` changes `KeystoreState` to "locked" state, which should trigger a route change in `WalletView, isn't?

File: src/services/wallet/types.ts (hunk 2 above)
Comment (veado) [reply src/services/wallet/types.ts:21]: We've decided to keep this change ^

File: src/components/wallet/UnlockForm.tsx [hunk 6]
```diff
@@ -83,6 +88,15 @@ const UnlockForm: React.FC<Props> = (props: Props): JSX.Element => {
     form.resetFields()
   }, [form])
 
+  const onRemove = useCallback(() => {
+    const isConfirmed = window.confirm('remove wallet?')
```
[the remark this answers is not in the document]
Comment (veado) [reply src/components/wallet/UnlockForm.tsx:92]: ^ Can be fixed later with #270.

File: src/components/wallet/UnlockForm.tsx [hunk 7]
```diff
@@ -113,6 +119,9 @@ const UnlockForm: React.FC<Props> = (props: Props): JSX.Element => {
 
   return (
     <Form form={form} onFinish={submitForm}>
+      <Modal visible={showRemoveModal} onCancel={hideRemoveConfirm} onOk={onRemoveConfirm}>
+        Remove wallet?
```
Comment (veado) [inline src/components/wallet/UnlockForm.tsx:123]: Use `ì18n`

File: src/components/wallet/UnlockForm.tsx (hunk 5 above)
Comment (thatStrangeGuyThorchain) [reply src/components/wallet/UnlockForm.tsx]: yep. u r right

**Pull request merged**